docs/guides/gcp-private-service-connect-workspace.md

```diff
@@ -0,0 +1,311 @@
+  region                       = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
+  public_access_enabled        = true
+  private_access_level         = "ACCOUNT"
+}
+
+resource "databricks_mws_workspaces" "this" {
+  provider       = databricks.accounts
+  account_id     = var.databricks_account_id
+  workspace_name = "tf-demo-test-${random_string.suffix.result}"
+  location       = google_compute_subnetwork.network-with-private-secondary-ip-ranges.region
+  cloud_resource_container {
+    gcp {
+      project_id = var.google_project
+    }
+  }
+
+  private_service_connect_id = databricks_mws_private_access_settings.pas.private_access_settings_id
+  network_id = databricks_mws_networks.this.network_id
+  gke_config {
+    connectivity_type = "PRIVATE_NODE_PUBLIC_MASTER"
+    master_ip_range   = "10.3.0.0/28"
+  }
+
+  token {
+    comment = "Terraform"
+  }
+
+  # this makes sure that the NAT is created for outbound traffic before creating the workspace
+  depends_on = [google_compute_router_nat.nat]
+}
+
+output "databricks_host" {
+  value = databricks_mws_workspaces.this.workspace_url
+}
+
+output "databricks_token" {
+  value     = databricks_mws_workspaces.this.token[0].token_value
+  sensitive = true
+}
+```
+
+### Data resources and Authentication is not configured errors
+
+*In Terraform 0.13 and later*, data resources have the same dependency resolution behavior [as defined for managed resources](https://www.terraform.io/docs/language/resources/behavior.html#resource-dependencies). Most data resources make an API call to a workspace. If a workspace doesn't exist yet, `default auth: cannot configure default credentials` error is raised. To work around this issue and guarantee a proper lazy authentication with data resources, you should add `depends_on = [databricks_mws_workspaces.this]` to the body. This issue doesn't occur if workspace is created *in one module* and resources [within the workspace](workspace-management.md) are created *in another*. We do not recommend using Terraform 0.12 and earlier, if your usage involves data resources.
+
+```hcl
+data "databricks_current_user" "me" {
+  depends_on = [databricks_mws_workspaces.this]
+}
+```
+
+## Provider configuration
+
+In [the next step](workspace-management.md), please use the following configuration for the provider:
+
+```hcl
+provider "databricks" {
+  host  = module.dbx_gcp.workspace_url
+  token = module.dbx_gcp.token_value
+}
+```
+
+We assume that you have a terraform module in your project that creats a workspace (using [Databricks Workspace](#creating-a-databricks-workspace) section) and you named it as `dbx_gcp` while calling it in the **main.tf** file of your terraform project. And `workspace_url` and `token_value` are the output attributes of that module. This provider configuration will allow you to use the generated token during workspace creation to authenticate to the created workspace.
```

Comment:
Unrelated to PSC. Do we need them?

Reply:
I copied it from gcp_workspace.md. I don't know what they are. Removed.